src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/PreferTypedStringBuilderAppendOverloadsTests.cs

```diff
@@ -0,0 +1,273 @@
+
+                class C
+                {
+                    public void M1() => new StringBuilder().Append(" + value + @");
+                    public void M2() => new StringBuilder().Insert(42, " + value + @");
+                }");
+        }
+
+        [Theory]
+        [InlineData("42")]
+        [InlineData("DayOfWeek.Monday")]
+        [InlineData("DateTime.Now")]
+        public async Task NoDiagnostic_FormattedToString(string value)
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+                using System;
+                using System.Globalization;
+                using System.Text;
+
+                class C
+                {
+                    public void M1()
+                    {
+                        new StringBuilder()
+                            .Append(" + value + @".ToString(""X4""))
+                            .Append(" + value + @".ToString(CultureInfo.CurrentCulture))
+                            .Append(" + value + @".ToString(""X4"", CultureInfo.CurrentCulture))
+                            .Append(((IFormattable)" + value + @").ToString(""X4"", CultureInfo.CurrentCulture))
+                            .Append(((IFormattable)" + value + @").ToString());
+                    }
+
+                    public void M2()
+                    {
+                        new StringBuilder()
+                            .Insert(1, " + value + @".ToString(""X4""))
+                            .Insert(1, " + value + @".ToString(CultureInfo.CurrentCulture))
+                            .Insert(1, " + value + @".ToString(""X4"", CultureInfo.CurrentCulture))
+                            .Insert(1, ((IFormattable)" + value + @").ToString(""X4"", CultureInfo.CurrentCulture))
+                            .Insert(1, ((IFormattable)" + value + @").ToString());
+                    }
+                }");
+        }
+
+        [Fact]
+        public async Task NoDiagnostic_NotRelevantMethod()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+                using System;
+                using System.Text;
+
+                class C
+                {
+                    public void M()
+                    {
+                        var sb = new StringBuilder();
+                        sb.AppendLine(42.ToString());
+                        sb.Replace(42.ToString(), ""42"");
+
+                        Console.WriteLine(42.ToString());
+
+                        Append(42.ToString());
+                    }
+
+                    private static void Append(string value) { }
+                    private static void Append(int value) { }
+                }");
+        }
+
+        [Fact]
+        public async Task CSharpFixer()
+        {
+            await VerifyCS.VerifyCodeFixAsync(@"
+                using System;
+                using System.Text;
+
+                class C
+                {
+                    public void M1() => new StringBuilder().Append([|42.ToString()|]);
```

Comment:
Should the squiggle be under the whole argument, or just the `.ToString()` part?  I'd put it just under ToString() since removing those characters is the recovery action.  Consider it as a complex flow:

```C#
public void M1() =>
    new StringBuilder.Append(
        new OtherThing
        {
            PropA = ...,
            PropB = ...,
            PropC = ...,
            PropD = ...,
        }.GetHashCode().ToString());
```

It doesn't need multiple lines of squiggle here.

Reply:
While this might be a lame argument, I did it under the whole expression as it makes the fixer simpler / easier to get right (and there didn't seem to be a significant usability benefit to just having the squiggle under the ToString()).  The fixer is then passed the whole relevant expression, rather than part of the expression it needs to walk up from to figure out the relevant part.

Reply:
So long as it was a conscious choice :)